pkg/api/api.go

```diff
@@ -20,12 +23,46 @@ type Response struct {
 	Checks        pkg.Checks    `json:"checks"`
 	ChecksSummary models.Checks `json:"checks_summary,omitempty"`
 }
-type DetailResponse struct {
+
+// Latency metrics in milliseconds
+type Latency struct {
+	P95 int `json:"p95,omitempty"`
+	P97 int `json:"p97,omitempty"`
+	P99 int `json:"p99,omitempty"`
+}
+
+func measureLatency(data []pkg.Timeseries) Latency {
```

Comment:
We need todo in the DB,   If there are 1000 statuses i am expecting 1000 passed and latency measurements on those 1000, and then the graph to show an aggregated view

Reply:
The latency is measured before aggregation so it's already like that.
Unless you mean to measure the latency in the SQL query itself?